utils/src/main/java/com/nedap/archie/json/DateTimeSerializer.java

```diff
@@ -20,7 +22,13 @@ public void serialize(TemporalAccessor temporalAccessor, JsonGenerator jsonGener
         if(temporalAccessor == null) {
             jsonGenerator.writeString("");
         }
-        jsonGenerator.writeString(DateTimeSerializerFormatters.ISO_8601_DATE_TIME.format(temporalAccessor));
+        if(!temporalAccessor.isSupported(ChronoField.HOUR_OF_DAY) &&
+                !temporalAccessor.isSupported(ChronoField.MINUTE_OF_HOUR) &&
+                !temporalAccessor.isSupported(ChronoField.SECOND_OF_MINUTE)) {
+            jsonGenerator.writeString(DateTimeSerializerFormatters.ISO_8601_DATE_TIME.format(temporalAccessor));
```

Comment:
The added tests are failing currently, probably because this line is exactly the same as the one in the else.

Reply:
Oh yes that was part of a test that I partially restored. Sorry. Fixed it!.